src/harness/sas_testcase.py

```diff
@@ -141,3 +141,24 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
     while not self._sas_admin.GetDailyActivitiesStatus()['completed']:
       time.sleep(10)
     signal.alarm(0)
+    
+  def TriggerFullActivityDumpAndWaitUntilComplete(self):
+    self._sas_admin.TriggerFullActivityDump()
+    request_time = datetime.utcnow()
+    
+    signal.signal(signal.SIGALRM,
+                  lambda signum, frame:
+                  (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
+    dump_message = self._sas.GetFullActivityDump()
+    dump_time = datetime.strptime(dump_message['generationDateTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+    # Timeout after 2 hours if it's not completed
+    signal.alarm(7200)
+    # Check the Status of Daily Activities every 10 seconds
+    while request_time > dump_time:
+      time.sleep(10)
+      dump_message = self._sas.GetFullActivityDump()
+      dump_time = datetime.strptime(dump_message['generationDateTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+    signal.alarm(0)
+    return dump_message
```

Comment:
Not necessary, I think?

Reply:
did you mean:  it will not be excuted after signal.alarm(0)? or it is better not to return it here and to get it again in the test?
because I return the dump message which is necessary for the test (line 275)

Reply:
I meant that it's better to get it again in the test. In other scenarios, we will have multiple SAS THs and they will each need to pull FAD from SAS UUT, so in that case we would iterate over the SAS THs. But I can see how this would be useful in a test case where you only have one SAS TH, so I'm OK to keep it as well: test cases with multiple SAS THs can always ignore the output.